flag_engine/engine.py

```diff
@@ -101,9 +102,11 @@ def _get_identity_feature_states_dict(
     identity_segments = get_identity_segments(environment, identity, override_traits)
     for matching_segment in identity_segments:
         for feature_state in matching_segment.feature_states:
-            # note that feature states are stored on the segment in descending priority
-            # order so we only care that the last one is added
-            # TODO: can we optimise this?
+            if feature_state.feature in feature_states:
+                if _should_not_override(
```

Comment:
I think perhaps we should be a bit more explicit? I tend to think a method on the FeatureStateModel would be better? Something like: 

```python
class FeatureStateModel:
    def is_higher_segment_priority(self, other):
        """ 
        return `True` if `self` is higher segment priority than `other` 
        (i.e. has lower value for feature_segment.priority)
        """
        ...
```

Then this section of code becomes: 

```python
existing_fs = feature_states[feature_state.feature]
if feature_state.is_higher_segment_priority(existing_fs):
    feature_states[feature_state.feature] = feature_state
```